src/main/scala/nodes/learning/KernelRidgeRegression.scala

```diff
@@ -0,0 +1,343 @@
+package nodes.learning
+
+import workflow.LabelEstimator
+
+import breeze.linalg._
+import breeze.math._
+import breeze.numerics._
+import org.apache.spark.rdd.RDD
+import org.apache.spark.broadcast.Broadcast
+import org.apache.spark.SparkConf
+import org.apache.spark.SparkContext
+import org.apache.spark.SparkContext._
+import utils._
+
+
+import edu.berkeley.cs.amplab.mlmatrix.RowPartitionedMatrix
+import edu.berkeley.cs.amplab.mlmatrix.util.{Utils => MLMatrixUtils}
+
+import scala.util.Random
+
+class KernelRidgeRegression(
+  kernelBlockGenerator: KernelGenerator,
```

Comment:
Its equivalent in the sense that you will get the same answer but its worse in terms of conditioning / computation costs. To make this a bit more clear in the least squares case we are solving `x = (A^T * A)^-1 (A^T * b)`
So here the condition number of `A` is squared and we need to spend time to compute `A^T * A`

But if A is a square matrix (like a kernel) then we can solve a linear system `x = A^-1 * b`. Now you can use the least squares method for the square matrix as well but that'll unnecessarily make the condition number worse and we'll need to spend a lot of FLOPs computing `A^T * A` (n^3 for a n*n matrix). 

TLDR is that for square matrices you can solve the linear system directly instead of using the least squares method